tests/functional/cli/test_reloader.py

```diff
@@ -21,26 +32,46 @@ def modify_file(filename, contents):
         f.write(contents)
 
 
-def assert_reload_happens(root_dir, when_modified_file):
-    http_thread = mock.Mock(spec=reloader.HTTPServerThread)
-    p = reloader.WorkerProcess(http_thread)
+def assert_reload_happens(root_dir, when_modified_file, using_worker_class):
+    http_thread = mock.Mock(spec=chalice.local.HTTPServerThread)
+    worker_cls = get_worker_cls(using_worker_class)
+    p = worker_cls(http_thread)
     modify_file_after_n_seconds(when_modified_file, 'contents')
     rc = p.main(root_dir, MAX_TIMEOUT)
-    assert rc == reloader.RESTART_REQUEST_RC
+    assert rc == chalice.cli.filewatch.RESTART_REQUEST_RC
+
+
+def get_worker_cls(worker_class_name):
+    if worker_class_name == 'watchdog':
+        if not WATCHDOG_AVAILABLE:
+            raise unittest.SkipTest("Test requires watchdog package.")
+        else:
+            return WatchdogWorkerProcess
+    elif worker_class_name == 'stat':
+        return StatWorkerProcess
+    else:
+        raise RuntimeError("Unknown worker class type name: %s"
+                           % worker_class_name)
 
 
-def test_can_reload_when_file_created(tmpdir):
+@use_all_watcher_types
+def test_can_reload_when_file_created(tmpdir, worker_class_type):
     top_level_file = str(tmpdir.join('foo'))
-    assert_reload_happens(str(tmpdir), when_modified_file=top_level_file)
+    assert_reload_happens(str(tmpdir), when_modified_file=top_level_file,
+                          using_worker_class=worker_class_type)
 
 
-def test_can_reload_when_subdir_file_created(tmpdir):
+@use_all_watcher_types
```

Comment:
Yeah I think it would be nice to flesh out some more tests here since we are rolling out our own solution. For example, it would be good to make sure we can handle file removals.